cirq/ops/controlled_gate.py

```diff
@@ -142,7 +142,20 @@ def _resolve_parameters_(self, param_resolver):
         return ControlledGate(new_sub_gate, self.control_qubits)
 
     def _trace_distance_bound_(self):
-        return protocols.trace_distance_bound(self.sub_gate)
+        if self._is_parameterized_():
+            return 1
+        if cirq.has_unitary(self.sub_gate):
+            angles = np.sort(
+                np.append(
+                    np.angle(np.linalg.eigvals(protocols.unitary(
+                        self.sub_gate))), 0))
+            maxim = 2 * np.pi + angles[0] - angles[-1]
+            for i in range(1, len(angles)):
+                maxim = max(maxim, angles[i] - angles[i - 1])
+            if maxim <= np.pi:
+                return 1
+            return np.sin(0.5 * maxim)
+        return 1
```

Comment:
Code copying.  Same comments as in controlled_operation.py.

Reply:
added utility method that controlled_gate imports to avoid the copying problem.